Sources/Rendering/Core/VolumeProperty/index.js

```diff
@@ -203,6 +203,30 @@ function vtkVolumeProperty(publicAPI, model) {
   publicAPI.getInterpolationTypeAsString = () =>
     macro.enumToString(InterpolationType, model.interpolationType);
 
+  publicAPI.getColorMixCode = () => {
+    if (model.customColorMixCode) {
+      return model.customColorMixCode;
+    }
+    switch (model.colorMixPreset) {
+      case ColorMixPreset.ADDITING:
+        return `
+          float opacity0 = goFactor.x * pwfValue0;
+          float opacity1 = goFactor.y * pwfValue1;
+          float opacitySum = opacity0 + opacity1;
+          if (opacitySum == 0.0) {
+            return vec4(0.0);
+          }
+          #if vtkLightComplexity > 0
+            applyLighting(tColor0, normal0);
+            applyLighting(tColor1, normal1);
+          #endif
+          vec3 mixedColor = mix(tColor0, tColor1, opacity1 / opacitySum);
+          return vec4(mixedColor, opacitySum);`;
```

Comment:
this WebGL code should not be in vtkVolumeProperty. Instead, it should go in the `OpenGL/VolumeMapper` file.